src/main/java/org/spongepowered/common/network/PacketUtil.java

```diff
@@ -166,6 +166,13 @@ private static boolean creativeCheck(Packet<?> packetIn, EntityPlayerMP playerMP
         return packetIn instanceof CPacketCreativeInventoryAction;
     }
 
+    /**
+     * Handles firing of events that are triggered before server logic. Be aware that this method makes changes to the current context
+     * of the current frame of the {@link CauseStackManager.StackFrame} (namely to set the used_item).
```

Comment:
Perhaps we should have a nice documented annotation of our own just to self document methods that are not stack neutral? It would help in self documentation in some cases (of course, something that we should be able to instantly see and understand that something is stack altering to the trained eye versus the untrained eye).